corpus/v2/custom-fields.json

```diff
@@ -0,0 +1,93 @@
+{
+  "name": "widget",
+  "abstract": "Widget for PostgreSQL",
+  "description": "¿A widget is just thing thing, yoŭ know?",
+  "version": "0.2.5",
+  "maintainers": [
+    {
+      "name": "David E. Wheeler",
+      "email": "david@justatheory.com",
+      "x_looking_for_jobs": false
+    }
+  ],
+  "license": "PostgreSQL",
+  "contents": {
+    "extensions": {
+      "pair": {
+        "sql": "sql/widget.sql",
+        "control": "widget.control"
+      }
+    },
+    "modules": {
+      "examples": {
+        "type": "extension",
+        "lib": "target/release/example",
+        "preload": "server",
+        "x_is_c_only": false
+      }
+    },
+    "x_contains_shared_objects": true
+  },
+  "resources": {
+    "homepage": "http://widget.example.org/",
+    "x_support_email": "widget@example.org",
+    "badges": [
+      {
+        "alt": "CI Status",
+        "src": "https://github.com/example/pg-widget/actions/workflows/ci.yml/badge.svg",
+        "url": "https://github.com/example/pg-widget/actions/workflows/ci.yml",
+        "x_updated_every": "1h"
+      }
+    ]
+  },
+  "dependencies": {
+    "postgres": {
+      "version": "14.0",
+      "X_RELEASE": "REL_14_BETA3"
+    },
+    "x_forbid_unsafe": true,
+    "packages": {
+      "run": {
+        "requires": {
+          "pkg:generic/python": "2.0",
```

Comment:
So would that be something like this?

```json
"packages": {
      "run": {
        "requires": {
          "pkg:generic/python": "2.0",
          "pkg:pypi/psycopg2": 0,
          "x_pkg:something"
        },
        "recommends": {
          "pkg:pgxn/pg_jobmon": "1.4.1",
          "x_job": "test"
        }
    }
```

Because that'd be a small issue since these fields are currently of the type `Option<HashMap<String, VersionRange>>`. If we want to add `x_`/`X_` support for those then I think it'd require some more custom serde code

Reply:
Would it work to add another variant of the VersionRange enum?

I'm fine to omit this one for now, though, and deal with it later if it comes up.

Reply:
Not sure! I think it'd have to be a new struct e.g. `VersionMap`:

```rust
struct VersionMap {
    versions: HashMap<String, VersionRange>,
    custom_props: HashMap<String, Value>,
}
```

with custom `Serialize` and `Deserialize`

Reply:
Let's skip it for now; let's remove [this line](https://github.com/pgxn/meta/blob/71afca09c5f516320700aba3922a0e5f797f014b/schema/v2/packages.schema.json#L36) so that it's not formally supported.